pkg/compose/publish.go

```diff
@@ -64,11 +64,29 @@ func (s *composeService) publish(ctx context.Context, project *types.Project, re
 		return err
 	}
 
-	resolver := imagetools.New(imagetools.Opt{
-		Auth: s.configFile(),
+	config := s.dockerCli.ConfigFile()
+
+	resolver := docker.NewResolver(docker.ResolverOptions{
+		Hosts: docker.ConfigureDefaultRegistries(
+			docker.WithAuthorizer(docker.NewDockerAuthorizer(
+				docker.WithAuthCreds(func(host string) (string, string, error) {
+					if host == "registry-1.docker.io" {
+						host = "https://index.docker.io/v1/"
+					}
+					auth, err := config.GetAuthConfig(host)
```

Comment:
Maybe this should use the internal utility to keep it central? Or is the logic different here? https://github.com/docker/compose/blob/032e0309ee09055d9bc563c63e19b197ed405c1c/internal/registry/registry.go#L30-L38


That one doesn't use `registry-1.docker.io` though; ISTR we had that at some point, but some users depended on `registry-1.docker.io` not to be normalised to `docker.io` (and thus "default auth")


(😭 I need to cry every time I run into these)